src/cpp/src/sequence_group.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string_view>
+#include "sequence_group.hpp"
+
+namespace ov {
+namespace genai {
+size_t Sequence::_make_hash(size_t content_length) {
+        auto sequence_group = get_sequence_group_ptr();
+        auto block_size = sequence_group->get_block_size();
+        size_t block_start_idx = content_length - (content_length % block_size);
+        if (block_start_idx == content_length) {
+            block_start_idx -= block_size;
+        }
+
+        // hash of current block depends on prefix hashes
+        std::vector<int64_t> content;
+        size_t prefix_hashes_needed_count = content_length % block_size == 0 ? content_length / block_size - 1 : content_length / block_size;
```

Comment:
is suppose it's the same as block_start_idx / block_size ?

Reply:
Yes, replaced with `block_start_idx / block_size`